main.js

```diff
@@ -2,19 +2,35 @@
 
 const fs = require('fs');
 const path = require('path');
+const glob = require('glob');
 
 const {readConfig, parseArgs} = require('./config');
 const {readFile} = require('./utils');
 const runner = require('./runner');
 const render = require('./render');
 const {testsVersion, pintfVersion} = require('./version');
 
-function load_tests(args, tests_dir) {
-    let test_names = (
-        fs.readdirSync(tests_dir)
-            .filter(n => n.endsWith('.js'))
-            .map(n => n.substring(0, n.length - '.js'.length))
-    );
+/**
+ * @param {*} args 
+ * @param {string} tests_dir 
+ * @param {string} [glob_pattern] 
+ */
+function load_tests(args, tests_dir, glob_pattern) {
+    let test_names = [];
+    
+    if (! glob_pattern) {
+        test_names = (
+            fs.readdirSync(tests_dir)
+                .filter(n => n.endsWith('.js'))
+                .map(n => n.substring(0, n.length - '.js'.length))
+        );
+    } else {
+        test_names = glob
+            .sync(glob_pattern, {
+                cwd: tests_dir,
+            })
+            .map(n => path.join(path.dirname(n), path.basename(n)));
```

Comment:
We should use a struct instead of testnames here.

Reply:
done 🎉